tests/tasks/test_testing.py

```diff
@@ -71,3 +79,82 @@ def test_dummy_timestamp_feature_task(dummy_eopatch: EOPatch):
+@pytest.mark.parametrize(
+    ("feature_type", "shape", "dtype", "min_value", "max_value"),
+    [
+        (FeatureType.DATA, (10, 20, 21, 5), np.float32, -3, -1.5),
+        (FeatureType.DATA_TIMELESS, (83, 69, 1), int, -2, 5),
+        (FeatureType.MASK, (10, 4, 16, 7), np.int8, -3, 7),
+        (FeatureType.LABEL, (10, 271), bool, 0, 1),
+    ],
+)
+@pytest.mark.parametrize("seed", [1, 22, 9182])
+def test_generate_raster_feature_task_uniform(dummy_eopatch, feature_type, shape, dtype, min_value, max_value, seed):
+    dummy_eopatch.timestamps = ["2011-08-12"] * 10
+    feature = feature_type, "FEATURE"
+    configuration = UniformDistribution(min_value, max_value)
+
+    task = GenerateRasterFeatureTask(feature, shape=shape, dtype=dtype, distribution=configuration)
+    eopatch = task.execute(dummy_eopatch.copy(), seed=seed)
+
+    assert feature in eopatch
+    assert len(eopatch.get_features()) == 3
+
+    data: np.ndarray = eopatch[feature]
+    assert data.shape == shape
+    assert data.dtype == dtype
+
+    # check that the data is uniform 'enough'
+    if is_discrete_type(data.dtype):
+        # bins dont work well with few int values
+        values, bin_nums = np.unique(data, return_counts=True)
+        assert np.amin(values) == min_value
+        assert np.amax(values) == max_value
+        threshold = data.size / len(bin_nums) * 0.9
+        assert all(bin_nums > threshold)
+
+    else:
+        assert np.amin(data) >= min_value
+        assert np.amax(data) <= max_value
+        bin_nums, _ = np.histogram(data, bins=5, range=(min_value, max_value))
+        threshold = data.size / 5 * 0.9
+        assert all(bin_nums > threshold)
+
+    assert eopatch == task.execute(dummy_eopatch.copy(), seed=seed), "Same seed yields different results!"
+    assert eopatch != task.execute(dummy_eopatch.copy(), seed=seed + 1), "Different seed yields same results!"
+
+
+@pytest.mark.parametrize(
+    ("feature_type", "shape", "dtype", "mean", "std"),
+    [
+        (FeatureType.DATA, (7, 20, 31, 1), np.float32, -3, 1.5),
+        (FeatureType.DATA_TIMELESS, (30, 61, 2), float, -3422.23, 1522),
+        (FeatureType.DATA_TIMELESS, (83, 69, 10), int, -2, 10),
+    ],
+)
+@pytest.mark.parametrize("seed", [1, 22, 9182])
+def test_generate_raster_feature_task_normal(dummy_eopatch, feature_type, shape, dtype, mean, std, seed):
+    dummy_eopatch.timestamps = ["2011-08-12"] * 7
+    feature = feature_type, "FEATURE"
+    configuration = NormalDistribution(mean, std)
+
+    task = GenerateRasterFeatureTask(feature, shape=shape, dtype=dtype, distribution=configuration)
+    eopatch = task.execute(dummy_eopatch.copy(), seed=seed)
+
+    assert feature in eopatch
+    assert len(eopatch.get_features()) == 3
+
+    data: np.ndarray = eopatch[feature]
+    assert data.shape == shape
+    assert data.dtype == dtype
+    if not is_discrete_type(dtype):
+        assert shapiro(data)[1] > 0.05, "Shapiro-Wilks test rejects that the data is normally distributed."
```

Comment:
if I understand correctly, the Shapiro test is a test for normality of the data. Perhaps you could use the [Kolmogorov-Smirnov](https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.kstest.html) test, which doesn't depend on the distribution, and then you could use it in both cases, for the normal one as well as the uniform one, and the test could be simplified.

for example:

```python
from scipy import norm, uniform, kstest

data_norm = norm.rvs(size=30*60, loc=1, scale=0.5)
data_uni = uniform.rvs(size=30*60, loc=0, scale=2)

# good
kstest(data_uni, stats.uniform.cdf, args=(0, 2))
Out[84]: KstestResult(statistic=0.01889668569539571, pvalue=0.5350380083686563, statistic_location=1.5489044825019025, statistic_sign=-1)

# bad
kstest(data_norm, stats.uniform.cdf, args=(0, 2))
Out[85]: KstestResult(statistic=0.10682627834303782, pvalue=2.4249614533423762e-18, statistic_location=1.4252363322028132, statistic_sign=1)

# bad
kstest(data_uni, stats.norm.cdf, args=(1, 0.5))
Out[86]: KstestResult(statistic=0.10891666539212586, pvalue=4.735882711535022e-19, statistic_location=1.5100065744155635, statistic_sign=-1)

# good
kstest(data_norm, stats.norm.cdf, args=(1, 0.5))
Out[87]: KstestResult(statistic=0.025362671109045776, pvalue=0.1938881590422188, statistic_location=1.3109111847328712, statistic_sign=1)
```

Reply:
switched to K-S for uniform and chi-square for the discrete uniform

Reply:
will try to join the tests as well

Reply:
joined, switched the normal distribution to k-s as well, so that we have more guarantee that the mean and std are correct.